docs/ABOUT.md

```diff
@@ -1,16 +1,9 @@
-PureScript is a purely functional, statically-typed programming language with
-type inference.
+PureScript is a purely functional, statically-typed programming language with global type inference.
 
-**Functional** means that functions are first-class data types.
+**Functional** means computation is modeled as the evaluation of expressions rather than the execution of a sequence of instructions.
 
-**Purely Functional** means (roughly) that there are no side effects.
-Every function will always return the same value for a given argument
-and will do nothing else.
+**Purely Functional** means that it is possible to tell if a computation does side-effects or not. Meaning that if a computation changes a global variable, read from input, write to a socket, etc. it will be reflected in the type of the computation.
```

Comment:
I would say "if a computation _has_ side-effects or not_, meaning that_" and "_reads_ from input, _writes_ to a socket"